### PR TITLE
Implement aws_huffman_get_encoded_length

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/huffman.c [hunk 1]
```diff
@@ -114,6 +114,30 @@ static int encode_write_bit_pattern(struct encoder_state *state, struct aws_huff
...
+    size_t num_bits = 0;
+
+    while (to_encode.len) {
+        uint8_t new_byte = 0;
+        aws_byte_cursor_read_u8(&to_encode, &new_byte);
+        struct aws_huffman_code code_point = encoder->coder->encode(new_byte, encoder->coder->userdata);
+        num_bits += code_point.num_bits;
+    }
+
+    size_t length = num_bits / 8;
```
Comment (JonathanHenson) [inline source/huffman.c]: just a thought:

how hot is this path? if it is called when we need to be fast, you can change these to bit operations:

num_bits >> 3

and 

if (num_bits & 0x07)

Comment (justinboswell) [reply source/huffman.c]: Check the disasm, I will bet you anything that compilers are already doing this.

Comment (ColdenCullen) [reply source/huffman.c]: BOY DO I HAVE NEWS FOR YOU https://godbolt.org/z/3df92J

tl;dr: the compiler does this for me, and I think the way it is now is easier to read.
